internal/pkg/override/override.go

```diff
@@ -0,0 +1,77 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package override defines functionality to interact with the "overrides/" directory
+// for accessing and mutating the Copilot generated AWS CloudFormation templates.
+package override
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/spf13/afero"
+)
+
+// Info holds metadata about an overrider.
+type Info int
+
+const (
+	unknownOverrider Info = iota
+	cdkOverrider
+	yamlPatchOverrider
+)
+
+// IsCDK returns true if the overrider is a CDK application.
+func (i Info) IsCDK() bool {
+	return i == cdkOverrider
+}
+
+// IsYAMLPatch returns true if the overrider is a YAML Patch document.
+func (i Info) IsYAMLPatch() bool {
+	return i == yamlPatchOverrider
+}
+
+// Lookup returns information indicating if the overrider is a CDK application or YAML Patch document.
+// If path does not exist or is an empty directory, then return an ErrNotExist.
+// If path is a YAML patch document, then IsYAMLPatch evaluates to true.
+// If path is a directory that contains a cdk.json file, then IsCDK evaluates to true.
+func Lookup(path string, fs afero.Fs) (Info, error) {
+	stat, err := fs.Stat(path)
+	if err != nil {
+		return unknownOverrider, &ErrNotExist{parent: err}
+	}
+
+	if stat.IsDir() {
```

Comment:
Correct me if I am wrong. Isn't yaml patch is also going to live under `overrides` directory? Just curious to know the reason behind putting `lookupCDK` inside `IsDir` if condition.